README.md

```diff
@@ -444,10 +444,132 @@ To build dense indexes (e.g., Faiss indexes) on your own document collections, f
 
 <details>
 <summary>Guide to indexing and searching English documents</summary>
+To build the dense index, Pyserini allows to either directly build Faiss Flat index via `pyserini.encode` with `output --to-faiss`, 
+or first encode collections into vectors via `pyserini.encode`, then build various types of Faiss index via `pyserini.index.faiss` based on the encoded collections. 
+ 
+<!-- ## How do I encode my own documents? -->
+<!-- In Pyserini, *encode* refers to the process that generating vectors from the text collections. -->
+<!-- These vectors can be used to perform retrieval directly, -->
+<!-- or to be further [*indexed*](usage-dense-indexes.md) into, for example, HNSW index. -->
+To use the `pyserini.encode`, the input should be in JSONL format. 
+Each line is a json dictionary containing two fields, i.e .`id` and `contents`.
+- `id` is the document id in string.
+- `contents` contains all the fields of the documents. By default, Pyserini expects the fields in contents are separated by `\n`.
+```json
+{
+  "id": "doc1",
+  "contents": "www.url.com\ntitle\nthis is the contents.\ndocument expansion"
+}
+```
+In the above example, the document has four fields in contents, `url`, `title`, `text` and `expand`.
+
+The field's boundary can be controled using `--delimiter` argument under `input`, see the example below.
+
+### Encode documents with Dense encoder
+```bash
+python -m pyserini.encode input   --corpus msmarco-passage-expanded \
```

Comment:
Change to our updated indentation style: https://github.com/castorini/pyserini/blob/master/docs/experiments-dpr.md

e.g.,

```
python -m pyserini.search.hybrid \
  dense  --index wikipedia-dpr-multi-bf \
         --encoded-queries dpr_multi-trivia-test \
  sparse --index wikipedia-dpr \
  fusion --alpha 0.95 \
  run    --topics dpr-trivia-test \
         --output runs/run.dpr.trivia-test.multi.bf.bm25.trec \
         --batch-size 36 --threads 12
```